handlers/validator.go

```diff
@@ -1177,7 +1177,7 @@ func ValidatorAttestations(w http.ResponseWriter, r *http.Request) {
 		totalCount = 0
 	}
 	lastAttestationEpoch := epoch
-	if ae.ExitEpoch != 9223372036854775807 {
+	if ae.ExitEpoch != 9223372036854775807 && ae.ExitEpoch < epoch {
 		lastAttestationEpoch = ae.ExitEpoch
 		totalCount = ae.ExitEpoch - ae.ActivationEpoch
 	}
```

Comment:
As discussed this would include the exit epoch in the attestations table for which no data exists from `GetValidatorAttestationHistory`.
As a result the first page for all validators that have already existed show only 9 entries.
Further pages are fine.

```suggestion
	if ae.ExitEpoch != 9223372036854775807 && ae.ExitEpoch <= epoch {
		lastAttestationEpoch = ae.ExitEpoch - 1
		totalCount = lastAttestationEpoch - ae.ActivationEpoch
	}
```

Here I am assuming for the count that you don't have any attestations in the ActivationEpoch.
If you do then +1.

Reply:
Thanks addapted the` <=` and `-1`  ... but I think `totalCount = ae.ExitEpoch - ae.ActivationEpoc` is correct